jupyterhub_traefik_proxy/toml.py

```diff
@@ -232,10 +236,11 @@ def get_target_data(d, to_find):
         """
         all_routes = {}
         async with self.mutex:
-            for key, value in self.routes_cache["frontends"].items():
-                escaped_routespec = "".join(key.split("_", 1)[1:])
-                routespec = escapism.unescape(escaped_routespec)
-                all_routes[routespec] = self._get_route_unsafe(routespec)
+            if self.routes_cache:
```

Comment:
What happens here `if not self.routes_cache`? When would that happen?

Reply:
I don't think we need it anymore. I ran into this problem when I had the 2 toml files in another directory (other than tests).
This situation can happen when TraefikTomlProxy finds an empty `rules.toml` file. When we start the proxy, we call _setup_static_config and this function checks the existence of `rules.toml` and if it finds it, it will load it into routes_cache. Since the file is empty, there will be no `frontends` or `backends` keys in the dict.